File: akka/actor-system_2.4.8/src/main/scala/demo/actor/scala/one/HelloActorDemo.scala
package demo.actor.scala.one

import akka.actor.{Actor, ActorRef, ActorSystem, Props}

/**
  * Simple Actor Which will print a String message.
  */
class HelloActor extends Actor { // Extending actor trait
    //  Receiving message
    def receive = {
        case msg: String => println("Message Received : " + msg + ", Sender Name : " + self.path.name)
        case _ => println("Unknown message") // Default case
    }
}

/**
  * Driver Program to Invoke An Actor
  *
  * @author Kumar Gauraw
  */
object HelloActorDemo {

    def main(args: Array[String]): Unit = {
        // Creating ActorSystem
        val actorSystem = ActorSystem("ActorSystem")
        //Creating actor
        val actor = actorSystem.actorOf(Props[HelloActor], "HelloActor")

        // Sending messages by using ! // ! or tell both can be used for tell
        actor ! "Hello Akka"
        actor ! 100.52
        //Sending messages by using tell
        actor.tell("Hello Actor!", ActorRef.noSender)

        //shutdown actor system
        actorSystem.terminate();
    }
}

File: akka/actor-system_2.4.8/src/main/java/demo/actor/one/HelloActorDemo.java
package demo.actor.one;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;
import akka.actor.UntypedActor;

/**
 * HelloActor is an Actor, created using java implementation.
 */
class HelloActor extends UntypedActor { //extended UntypedActor
    // Implemented onReceive
    public void onReceive(Object message) throws Throwable {
        if (message instanceof String) {
            System.out.println(message);
        } else {
            System.out.println("Unknown Message");
        }
    }
}


/**
 * Driver Program which create an Actor System, Instantiate HelloActor Actor
 * and send some message using actor (HelloActor) reference.
 *
 * @author Kumar Gauraw
 */
public class HelloActorDemo {

    public static void main(String[] args) {

        // Creating ActorSystem
        ActorSystem system = ActorSystem.create("ActorSystem");

        //Creating actor
        ActorRef helloActor = system.actorOf(Props.create(HelloActor.class));

        // Sending messages using tell()
        helloActor.tell("Hello Akka!", ActorRef.noSender());
        helloActor.tell("How Are You?", ActorRef.noSender());
        helloActor.tell(100, ActorRef.noSender());

        //shut down actor system.
        //system.shutdown(); //this method is deprecated
        system.terminate();
    }
}
